pyresample/geometry.py

```diff
@@ -318,14 +324,50 @@ def get_bbox_lonlats(self, vertices_per_side: Optional[int] = None, force_clockw
             warnings.warn("The `frequency` argument is pending deprecation, use `vertices_per_side` instead",
                           PendingDeprecationWarning, stacklevel=2)
         vertices_per_side = vertices_per_side or frequency
-        lons, lats = self._get_bbox_elements(self.get_lonlats, vertices_per_side)
+        if self.is_geostationary:
+            lon_sides, lat_sides = self._get_geo_boundary_sides(vertices_per_side=vertices_per_side)
+        else:
+            lon_sides, lat_sides = self._get_boundary_sides(self.get_lonlats, vertices_per_side)
```

Comment:
Why not put this `if self.is_geostationary` check into `self._get_boundary_sides` so that any method/function/user wanting the boundary sides would get usable output? Or are geostationary boundaries such a special case that that doesn't work or make sense?

Reply:
yes, good idea!

Reply:
Good idea to change @djhoese @mraspaud . I made the change however there are couple of decision to be take into consideration (to avoid changes in behaviour of some other functions relying on `_get_boundary_sides`. See comment now in `_get_boundary_sides`